mcs_new_builds/analysis.py

```diff
@@ -0,0 +1,195 @@
+# %% [markdown]
+# ### Imports and setup
+
+# %%
+from asf_core_data import get_mcs_installations, load_preprocessed_epc_data
+
+import pandas as pd
+import numpy as np
+
+# %%
+# change path to your own local version of EPC data
+epc_path = "/Users/chris.williamson/Documents/ASF_data"
+
+# %%
+mcs = get_mcs_installations("full")
+
+# %%
+# convert date columns to datetime type
+mcs["commission_date"] = pd.to_datetime(mcs["commission_date"])
+mcs["INSPECTION_DATE"] = pd.to_datetime(mcs["INSPECTION_DATE"])
+
+# %%
+# merge installation type columns and filter to domestic installations
+mcs["installation_type"] = mcs["installation_type"].fillna(
+    mcs["end_user_installation_type"]
+)
+mcs = mcs.loc[mcs.installation_type == "Domestic"].reset_index(
+    drop=True
+)
+
+# %% [markdown]
+# ### What proportion of domestic records in the MCS database relate to new-build installations?
+
+# %%
+# UPRNs of properties that have an EPC labelled as "new dwelling" (not necessarily the first one - see below)
+new_uprns = [uprn for uprn in mcs.loc[mcs["TRANSACTION_TYPE"] == "new dwelling"]["UPRN"] if uprn != "unknown"]
+
+# %%
+# filter to first records
+# records not linked to an EPC are kept
+first_records = (
+    mcs
+    .sort_values("INSPECTION_DATE")
+    .groupby("original_mcs_index")
+    .head(1)
+    .sort_values("original_mcs_index")
+)
+
+# %%
+# find number of days between first recorded EPC inspection and HP commission
+# could instead use the first EPC that labels the property as a "new dwelling", rather than the first overall - which is best? (see below)
+first_records["diff_epc_to_mcs"] = (
+    first_records["commission_date"] - first_records["INSPECTION_DATE"]
+).dt.days
+
+# assume dwelling was built with HP if:
+# - it has an EPC indicating that it is a new dwelling
+# - time difference between first EPC inspection and HP installation is less than 1 year
+first_records["assumed_hp_when_built"] = (
+    first_records["UPRN"].isin(new_uprns)
+) & (first_records["diff_epc_to_mcs"] < 365)
```

Comment:
if you apply my suggestion above then I would do something like below to make sure inspection isn't happening before commissioning the HP (although I am not sure we can trust the inspection dates to be this strict)
```suggestion
first_records["assumed_hp_when_built"] = (
    first_records["UPRN"].isin(new_uprns)
) & (first_records["diff_epc_to_mcs"] <= 365)
& (first_records["diff_epc_to_mcs"] >=0)
```

Reply:
With new builds I'm not actually sure whether the EPC inspection can only happen when the property is fully complete. I can imagine situations where the EPC inspector might just base the assessment on the builder's specifications or a similar property that is complete, so the commissioning date would end up being after the inspection date. But if the commissioning date is too long after the inspection date then that suggests the HP was installed later rather than being there where the property was built, which is why I've put a (slightly arbitrary) cutoff of 365 days here. The negatives should definitely be included as this is what I'd expect to happen most of the time - HP gets commissioned, then EPC inspection happens when the house is finished. There's no cutoff on this side since I didn't want to make any assumptions about how long it takes to build a property. Does that sound sensible?

Reply:
yep, that does sound super sensible!